core/node/base_token_adjuster/src/base_token_ratio_persister.rs

```diff
@@ -54,11 +82,86 @@ impl BaseTokenRatioPersister {
+                .await?;
+
+            result = self
+                .send_ratio_to_l1(new_ratio, gas_price, base_fee_per_gas, priority_fee_per_gas)
+                .await;
+            if let Some(err) = result.as_ref().err() {
+                tracing::info!(
+                    "Failed to update base token multiplier on L1, attempt {}, gas_price {}, base_fee_per_gas {}, priority_fee_per_gas {}: {}",
+                    attempt + 1,
+                    gas_price,
+                    base_fee_per_gas,
+                    priority_fee_per_gas,
+                    err
+                );
+                tokio::time::sleep(sleep_duration).await;
+                prev_base_fee_per_gas = Some(base_fee_per_gas);
+                prev_priority_fee_per_gas = Some(priority_fee_per_gas);
+            } else {
+                tracing::info!(
+                    "Updated base token multiplier on L1: numerator {}, denominator {}, gas_price {}, base_fee_per_gas {}, priority_fee_per_gas {}",
+                    new_ratio.numerator.get(),
+                    new_ratio.denominator.get(),
+                    gas_price,
+                    base_fee_per_gas,
+                    priority_fee_per_gas
+                );
+                return result;
+            }
+        }
+        result
+    }
+
+    async fn get_eth_fees(
+        &self,
+        prev_base_fee_per_gas: Option<u64>,
+        prev_priority_fee_per_gas: Option<u64>,
+    ) -> anyhow::Result<(u64, u64, u64)> {
+        // Use get_blob_tx_base_fee here instead of get_base_fee to optimise for fast inclusion.
+        // get_base_fee will cause the transaction to be stuck in the mempool for 10+ minutes.
+        let mut base_fee_per_gas = self.gas_adjuster.as_ref().get_blob_tx_base_fee();
+        let mut priority_fee_per_gas = self.gas_adjuster.as_ref().get_priority_fee();
+        if let Some(x) = prev_priority_fee_per_gas {
+            // Increase `priority_fee_per_gas` by at least 20% to prevent "replacement transaction under-priced" error.
+            priority_fee_per_gas = max(priority_fee_per_gas, (x * 6) / 5 + 1);
+        }
+
+        if let Some(x) = prev_base_fee_per_gas {
+            // same for base_fee_per_gas but 10%
+            base_fee_per_gas = max(base_fee_per_gas, x + (x / 10) + 1);
+        }
+
+        // Extra check to prevent sending transaction will extremely high priority fee.
+        if priority_fee_per_gas > self.config.max_acceptable_priority_fee_in_gwei {
+            panic!(
+                "Extremely high value of priority_fee_per_gas is suggested: {}, while max acceptable is {}",
+                priority_fee_per_gas,
+                self.config.max_acceptable_priority_fee_in_gwei
+            );
+        }
+
+        let gas_price = (*self.eth_client)
+            .as_ref()
+            .get_gas_price()
+            .await
+            .with_context(|| "failed getting gas price")?
+            .as_u64()
+            * 2;
```

Comment:
Why do you need gas price? It seems you use eip1559 txs so it's useless

Reply:
Good point - removed